### PR TITLE
Add regex support to DPLRouteMatcher

### PR DESCRIPTION
**Pull request by chrismaddern**
- Full regex support
- Existing functionality migrated to regex
- Wildcard routes supported with `@".*"`

### CODE REVIEW

**Comment by chrismaddern** [comment]
@wessmith @zimmpatrick since you guys worked on the host stuff, would love any thoughts before merging...

Idea is to replace our manual matching with a more Robust regex implementation.

**Comment by chrismaddern** [comment]
@wessmith LGTY?

File: DeepLinkSDK/RouteMatcher/DPLRouteMatcher.m [hunk 1]
```diff
@@ -2,10 +2,14 @@
 #import "DPLDeepLink_Private.h"
 #import "NSString+DPLTrim.h"
 
+static NSString * const DPLParameterRoutePattern = @":[a-zA-Z0-9-_]+";
+static NSString * const DPLParameterURLPattern = @"([a-zA-Z0-9-_]+)";
```
Comment (wessmith) [inline DeepLinkSDK/RouteMatcher/DPLRouteMatcher.m]: I think naming these:
`DPLRouteParameterPattern`
`DPLURLParameterPattern`

would make them more readable

File: DeepLinkSDK/RouteMatcher/DPLRouteMatcher.m [hunk 2]
```diff
@@ -23,66 +27,68 @@ - (instancetype)initWithRoute:(NSString *)route {
...
-            routeParameters[key] = pathComponent;
-        }
-        else if ([pathComponent isEqualToString:routeComponent]) {
-            isMatch = YES;
-        }
-        else {
-            isMatch = NO;
-            *stop   = YES;
+    for (NSTextCheckingResult *result in matches) {
+        for (int i = 1; i < result.numberOfRanges && i <= self.routeParamaterNames.count; i++) {
```
Comment (wessmith) [inline DeepLinkSDK/RouteMatcher/DPLRouteMatcher.m]: this makes me :frowning:

**Pull request merged**